cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/OrTrigger.java

```diff
@@ -46,4 +49,36 @@ public boolean isSatisfied(ProgramSchedule schedule, List<Notification> notifica
   public List<TriggerInfo> getTriggerInfos(TriggerInfoContext context) {
     return getUnitTriggerInfosAddRuntimeArgs(context);
   }
+
+  @Nullable
+  @Override
+  public AbstractCompositeTrigger getTriggerWithDeletedProgram(ProgramId programId) {
+    List<SatisfiableTrigger> updatedTriggers = new ArrayList<>();
+    for (Trigger trigger : getTriggers()) {
+      if (trigger instanceof ProgramStatusTrigger &&
+        programId.equals(((ProgramStatusTrigger) trigger).getProgramId())) {
+        // this program status trigger will never be satisfied, skip adding it to updatedTriggers
+        continue;
+      }
+      if (trigger instanceof AbstractCompositeTrigger) {
+        Trigger updatedTrigger = ((AbstractCompositeTrigger) trigger)
+          .getTriggerWithDeletedProgram(programId);
+        if (updatedTrigger == null) {
+          // the updated composite trigger will never be satisfied, skip adding it to updatedTriggers
+          continue;
+        }
+        // add the updated composite trigger into updatedTriggers
+        updatedTriggers.add((SatisfiableTrigger) updatedTrigger);
+      } else {
+        // the trigger is not a composite trigger, add it to updatedTriggers directly
+        updatedTriggers.add((SatisfiableTrigger) trigger);
+      }
+    }
+    // if the updatedTriggers is empty, the OR trigger will never be satisfied
+    if (updatedTriggers.isEmpty()) {
+      return null;
+    }
+    // return a new OR trigger constructed from the updated triggers
+    return new OrTrigger(updatedTriggers.toArray(new SatisfiableTrigger[updatedTriggers.size()]));
```

Comment:
similar comment to And. The constructor should take a list and it shouldn't return an Or if there's just one trigger.